client/src/pages/Balance/ReactStripe.js

```diff
@@ -0,0 +1,304 @@
+    <svg width="32px" height="32px" viewBox="0 0 32 32">
+      <path
+        fill="#FFF"
+        d="M15,7.05492878 C10.5000495,7.55237307 7,11.3674463 7,16 C7,20.9705627 11.0294373,25 16,25 C20.9705627,25 25,20.9705627 25,16 C25,15.3627484 24.4834055,14.8461538 23.8461538,14.8461538 C23.2089022,14.8461538 22.6923077,15.3627484 22.6923077,16 C22.6923077,19.6960595 19.6960595,22.6923077 16,22.6923077 C12.3039405,22.6923077 9.30769231,19.6960595 9.30769231,16 C9.30769231,12.3039405 12.3039405,9.30769231 16,9.30769231 L16,12.0841673 C16,12.1800431 16.0275652,12.2738974 16.0794108,12.354546 C16.2287368,12.5868311 16.5380938,12.6540826 16.7703788,12.5047565 L22.3457501,8.92058924 L22.3457501,8.92058924 C22.4060014,8.88185624 22.4572275,8.83063012 22.4959605,8.7703788 C22.6452866,8.53809377 22.5780351,8.22873685 22.3457501,8.07941076 L22.3457501,8.07941076 L16.7703788,4.49524351 C16.6897301,4.44339794 16.5958758,4.41583275 16.5,4.41583275 C16.2238576,4.41583275 16,4.63969037 16,4.91583275 L16,7 L15,7 L15,7.05492878 Z M16,32 C7.163444,32 0,24.836556 0,16 C0,7.163444 7.163444,0 16,0 C24.836556,0 32,7.163444 32,16 C32,24.836556 24.836556,32 16,32 Z"
+      />
+    </svg>
+  </button>
+);
+
+const useStyles = makeStyles({
+  text: {
+    fontSize: "3vw",
+    fontWeight: "800"
+  },
+  button: {
+    backgroundColor: "#43DDC1",
+    marginTop: "1vh",
+    marginBottom: "1vh",
+    width: "75%"
+  },
+  link: {
+    textDecoration: "none"
+  }
+});
+
+const CheckoutForm = ({ setAddCredit, credits }) => {
+  const { user } = useContext(UserContext);
+  const classes = useStyles();
+  const stripe = useStripe();
+  const elements = useElements();
+  const [error, setError] = useState(null);
+  const [cardComplete, setCardComplete] = useState(false);
+  const [processing, setProcessing] = useState(false);
+  const [paymentMethod, setPaymentMethod] = useState(null);
+  const [billingDetails, setBillingDetails] = useState({ name: "" });
+  const [madeSuccessfulPayment, setMadeSuccessfulPayment] = useState(false);
+
+  const handleSubmit = async event => {
+    event.preventDefault();
+    if (!stripe || !elements) {
+      // Stripe.js has not loaded yet. Make sure to disable
+      // form submission until Stripe.js has loaded.
+      return;
+    }
+
+    if (error) {
+      elements.getElement("card").focus();
+      return;
+    }
+
+    if (cardComplete) {
+      setProcessing(true);
+    }
+
+    const requestPaymentIntent = await axios.post(
+      `/user/${user._id}/purchase-credit`,
+      {
+        credits
+      }
+    );
+
+    const payload = await stripe.createPaymentMethod({
+      type: "card",
+      card: elements.getElement(CardElement),
+      billing_details: billingDetails
+    });
+
+    const confirm = await stripe.confirmCardPayment(
+      `${requestPaymentIntent.data.clientSecret}`,
+      {
+        payment_method: payload.paymentMethod.id
+      }
+    );
+
+    if (confirm.paymentIntent.status === "succeeded") {
```

Comment:
When payment fails, confirm.paymentIntent is undefined and the app crashes.
Tested with a couple of the card numbers https://stripe.com/docs/testing#cards that are meant to get a specific error response.